quic/s2n-quic-core/src/path/mtu.rs

```diff
@@ -232,11 +231,102 @@ impl Display for MtuError {
 #[cfg(feature = "std")]
 impl std::error::Error for MtuError {}
 
+/// Information about the path that may be used when generating or
+/// validating MTU configuration.
+#[non_exhaustive]
+pub struct PathInfo<'a> {
+    pub remote_address: event::api::SocketAddress<'a>,
+}
+
+impl<'a> PathInfo<'a> {
+    #[inline]
+    #[doc(hidden)]
+    pub fn new(remote_address: &'a inet::SocketAddress) -> Self {
+        PathInfo {
+            remote_address: remote_address.into_event(),
+        }
+    }
+}
+
+/// MTU configuration manager.
+#[derive(Copy, Clone, Debug)]
```

Comment:
I'd rather keep this `Clone`-only so we can add to it if needed

```suggestion
#[derive(Clone, Debug)]
```